debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -408,6 +414,13 @@ protected void handleGtidEvent(Event event) {
         String gtid = gtidEvent.getGtid();
         gtidSet.add(gtid);
         source.startGtid(gtid, gtidSet.toString()); // rather than use the client's GTID set
+        ignoreDmlEventByGtidSource = false;
+        if (gtidDmlSourceFilter != null && gtid != null) {
+            String uuid = gtid.trim().substring(0, gtid.indexOf(":"));
+            if (!gtidDmlSourceFilter.test(uuid)) {
```

Comment:
Good idea. If the config values are non-regex, we can use a `Set`, otherwise fall back to the `Predicate` filter?

How will we determine if the config values are non-regex? Just see if all the characters in the config are alphanumeric or commas?